javascript/ql/lib/semmle/javascript/security/dataflow/PrototypePollutingAssignmentCustomizations.qll

```diff
@@ -68,7 +68,7 @@ module PrototypePollutingAssignment {
   /**
    * A parameter of an exported function, seen as a source prototype-polluting assignment.
    */
-  class ExternalInputSource extends Source, DataFlow::SourceNode {
```

Comment:
Wouldn't it make more sense to treat `ReflectiveParmaetersNode` as a `SourceNode`?

Reply:
The `ReflectiveParametersNode` is a source node (but when I try to put that into the `extends` clause I get a non-monotonic recursion).  

But the source should still not be a `SourceNode`, because the `getALibraryInputParameter()` predicate does not return a `ReflectiveParametersNode`, that predicate returns either a parameter or a reference to the `arguments` variable.   
The location of `ReflectiveParametersNode` looks bad as a source (the location is the entire function), so I avoid using `ReflectiveParametersNode` directly as a source.